File: solutions/elixir/resistor-color-duo/lib/resistor_color_duo.ex
defmodule ResistorColorDuo do
  @values [:black, :brown, :red, :orange, :yellow, :green, :blue, :violet, :grey, :white]

  @doc """
  Calculate a resistance value from two colors
  """
  @spec value(colors :: [atom]) :: integer
  def value([tens | [ones | _]]),
    do: Enum.find_index(@values, &(&1 == tens)) * 10 + Enum.find_index(@values, &(&1 == ones))

  def value([single | _]), do: Enum.find_index(@values, &(&1 == single))
  def value([]), do: 0
end

File: solutions/go/resistor-color-duo/resistor_color_duo.go
package resistorcolorduo

// Better Be Right Or Your Great Big Values Go Wrong
var values = [10]string{
	"black", "brown", "red", "orange", "yellow",
	"green", "blue", "violet", "grey", "white"}

// Value should return the resistance value of a resistor with a given colors.
func Value(colors []string) (result int) {
	if len(colors) < 2 {
		return -1
	}

	for i, value := range values {
		if colors[0] == value {
			result += i * 10
		}
		if colors[1] == value {
			result += i
		}
	}
	return
}
